src/interaction/joystickcamerastates.cpp

```diff
@@ -160,87 +215,189 @@ void JoystickCameraStates::updateStateFromInput(const InputState& inputState,
     else {
         _localRotationState.velocity.decelerate(deltaTime);
     }
-
-    for (int i = 0; i < JoystickInputState::MaxButtons; ++i) {
-        auto itRange = _buttonMapping.equal_range(i);
-        for (auto it = itRange.first; it != itRange.second; ++it) {
-            bool active = global::joystickInputStates->button(i, it->second.action);
-
-            if (active) {
-                global::scriptEngine->queueScript(
-                    it->second.command,
-                    scripting::ScriptEngine::RemoteScripting(it->second.synchronization)
-                );
-            }
-        }
-    }
 }
 
-void JoystickCameraStates::setAxisMapping(int axis, AxisType mapping,
+void JoystickCameraStates::setAxisMapping(const std::string& joystickName,
+                                          int axis, AxisType mapping,
                                           AxisInvert shouldInvert,
-                                          AxisNormalize shouldNormalize,
+                                          JoystickType joystickType,
                                           bool isSticky,
                                           double sensitivity)
 {
     ghoul_assert(axis < JoystickInputState::MaxAxes, "axis must be < MaxAxes");
 
-    _axisMapping[axis].type = mapping;
-    _axisMapping[axis].invert = shouldInvert;
-    _axisMapping[axis].normalize = shouldNormalize;
-    _axisMapping[axis].isSticky = isSticky;
-    _axisMapping[axis].sensitivity = sensitivity;
+    JoystickCameraState* joystickCameraState = findOrAddJoystickCameraState(joystickName);
+    if (!joystickCameraState) {
+        return;
+    }
+
+    joystickCameraState->axisMapping[axis].type = mapping;
+    joystickCameraState->axisMapping[axis].invert = shouldInvert;
+    joystickCameraState->axisMapping[axis].joystickType = joystickType;
+    joystickCameraState->axisMapping[axis].isSticky = isSticky;
+    joystickCameraState->axisMapping[axis].sensitivity = sensitivity;
+
+    joystickCameraState->prevAxisValues[axis] =
+        global::joystickInputStates->axis(joystickName, axis);
+}
+
+void JoystickCameraStates::setAxisMappingProperty(const std::string& joystickName,
+                                                  int axis,
+                                                  const std::string& propertyUri,
```

Comment:
Since the propertyUri is only copied into the JoystickCameraState, it would be better to take that parameter as a `std::string` instead and then `std::move` out of it rather than making a copy